returnn/tf/layers/basic.py

```diff
@@ -6533,8 +6558,73 @@ def transform_input(
                 if i >= num_batch_dims and i != input_data.feature_dim_axis
             ]
             cls._check_defined_in_spatial_dims(len(in_spatial_dims) <= 1)
+
+        if use_time_mask:
+            input_data = input_data.copy()
+            x = input_data.placeholder
+            for dim in in_spatial_dims:
+                if not dim.need_masking():
+                    continue
+                axis = input_data.get_axis_from_description(dim)
+                mask = input_data.get_sequence_mask_broadcast(axis=axis)
+                x = tf_util.where_bc(mask, x, 0.0)
+
+            input_data.placeholder = x
+
         return input_data, num_batch_dims
 
+    @classmethod
+    def get_input_placeholder_with_same_static_padding(
+        cls,
+        input_data,
+        num_batch_dims,
+        filter_size,
+        strides,
+        out_batch_feature_major,
```

Comment:
Types not specified. I assume like this?
```suggestion
        input_data: Data,
        num_batch_dims: int,
        filter_size: Sequence[int],
        strides: Sequence[int],
        out_batch_feature_major: bool,
```
?

Reply:
ah yes, this is correct